core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -248,7 +248,7 @@ public void runInContext() {
     boolean deadlineExceeded = effectiveDeadline != null && effectiveDeadline.isExpired();
     if (!deadlineExceeded) {
       logIfContextNarrowedTimeout(
-          effectiveDeadline, callOptions.getDeadline(), context.getDeadline());
```

Comment:
This is a bug not caught by tests! But I'm not so bothered without test coverage because it's not convenient to test logging.

Reply:
I started to add a test, but then trashed it because I really don't want to start having tests for debug log statements.